File: Source/Geometry/closest_point.js
class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
}

function calculateDistance(a, b) {
    return Math.sqrt(Math.pow(b.x - a.x, 2) + Math.pow(b.y - a.y, 2));
}

function findClosestPoint(givenPoint, points) {
    let minDistance = calculateDistance(givenPoint, points[0]);
    let closestPoint = points[0];
    
    for(let i = 1; i < points.length; i++) {
        let distance = calculateDistance(givenPoint, points[i]);
        if(distance < minDistance) {
            minDistance = distance;
            closestPoint = points[i];
        }
    }
    
    return closestPoint;
}

File: Source/Geometry/closest_point.c
#include <math.h>

typedef struct {
    double x;
    double y;
} Point;

double calculateDistance(Point a, Point b) {
    return sqrt(pow(b.x - a.x, 2) + pow(b.y - a.y, 2));
}

Point findClosestPoint(Point givenPoint, Point points[], int size) {
    double minDistance = calculateDistance(givenPoint, points[0]);
    Point closestPoint = points[0];

    for(int i = 1; i < size; i++) {
        double distance = calculateDistance(givenPoint, points[i]);
        if(distance < minDistance) {
            minDistance = distance;
            closestPoint = points[i];
        }
    }

    return closestPoint;
}
